### DISCUSSION TITLE
Setting a custom Docker build context path?

### DISCUSSION THREAD

**Discussion by spiffytech** [dokku/dokku#6913] [Q&A] [answered]
I'm deploying a repo where my app is not at the repo root. I've configured Dokku with a custom Dockerfile path:

```sh
dokku builder-dockerfile:set rsl-dev dockerfile-path app-123/Dockerfile
```

However, the `git push` receive hook fails to build the container. It appears that Dokku is setting the build context to `.` (the repo root), while all the paths inside the Dockerfile are relative to the `app-123/` directory, since that's where the Dockerfile is.

How can I configure Dokku to set the build context to `app-123/` instead? (akin to `docker build app-123/`)

**Comment by josegonzalez** [comment, marked as answer]
You will need to set the `build-dir` builder property:

```
dokku builder:set rsl-dev build-dir app-123
```

This will change the root context to the `app-123` directory and execute your build commands there.
